Pipfile.lock

```diff
@@ -354,144 +389,175 @@
-                "sha256:a4a2cbcfc4cbf45cd126f531dedda8485671545b43107ded25ce952aac6fb308",
-                "sha256:b7fbfabdbcc78c4f6fc4712544b9b0d6bf171069c6e0e3cb82440dd10ced3406",
-                "sha256:c05b9e4fb1d8a41d41dec8786c94f3b95d3c5f528298d769eb8e73d293abc48d",
-                "sha256:d7df6eddb6054d21ca4d3c6249cae5578cb4602951fd2b6ee2f5510ffb098707",
-                "sha256:e0b61738ab504e656d1fe4ff0c0601387a5489ca122d55390ade31f9ca0e252d",
-                "sha256:eff7d4a85e9eea55afa34888dfeaccde99e7520b51f867ac28a48492c0b1130c",
-                "sha256:f05644db6779387ccdb468cc47a44b4356fc2ffa9287135d05b70a98dc83b89a"
-            ],
-            "markers": "python_version >= '3.5' and python_version < '3.8'",
-            "version": "==0.782"
+                "sha256:088cd9c7904b4ad80bec811053272986611b84221835e079be5bcad029e79dd9",
+                "sha256:0aadfb2d3935988ec3815952e44058a3100499f5be5b28c34ac9d79f002a4a9a",
+                "sha256:119bed3832d961f3a880787bf621634ba042cb8dc850a7429f643508eeac97b9",
+                "sha256:1a85e280d4d217150ce8cb1a6dddffd14e753a4e0c3cf90baabb32cefa41b59e",
+                "sha256:3c4b8ca36877fc75339253721f69603a9c7fdb5d4d5a95a1a1b899d8b86a4de2",
+                "sha256:3e382b29f8e0ccf19a2df2b29a167591245df90c0b5a2542249873b5c1d78212",
+                "sha256:42c266ced41b65ed40a282c575705325fa7991af370036d3f134518336636f5b",
+                "sha256:53fd2eb27a8ee2892614370896956af2ff61254c275aaee4c230ae771cadd885",
+                "sha256:704098302473cb31a218f1775a873b376b30b4c18229421e9e9dc8916fd16150",
+                "sha256:7df1ead20c81371ccd6091fa3e2878559b5c4d4caadaf1a484cf88d93ca06703",
+                "sha256:866c41f28cee548475f146aa4d39a51cf3b6a84246969f3759cb3e9c742fc072",
+                "sha256:a155d80ea6cee511a3694b108c4494a39f42de11ee4e61e72bc424c490e46457",
+                "sha256:adaeee09bfde366d2c13fe6093a7df5df83c9a2ba98638c7d76b010694db760e",
+                "sha256:b6fb13123aeef4a3abbcfd7e71773ff3ff1526a7d3dc538f3929a49b42be03f0",
+                "sha256:b94e4b785e304a04ea0828759172a15add27088520dc7e49ceade7834275bedb",
+                "sha256:c0df2d30ed496a08de5daed2a9ea807d07c21ae0ab23acf541ab88c24b26ab97",
+                "sha256:c6c2602dffb74867498f86e6129fd52a2770c48b7cd3ece77ada4fa38f94eba8",
+                "sha256:ceb6e0a6e27fb364fb3853389607cf7eb3a126ad335790fa1e14ed02fba50811",
+                "sha256:d9dd839eb0dc1bbe866a288ba3c1afc33a202015d2ad83b31e875b5905a079b6",
+                "sha256:e4dab234478e3bd3ce83bac4193b2ecd9cf94e720ddd95ce69840273bf44f6de",
+                "sha256:ec4e0cd079db280b6bdabdc807047ff3e199f334050db5cbb91ba3e959a67504",
+                "sha256:ecd2c3fe726758037234c93df7e98deb257fd15c24c9180dacf1ef829da5f921",
+                "sha256:ef565033fa5a958e62796867b1df10c40263ea9ded87164d67572834e57a174d"
+            ],
+            "markers": "python_version >= '3.9'",
```

Comment:
Oops, was this intentional? It breaks `pytest-mypy` when trying to run `pytest` with `python3.8`

Reply:
@bryant-finney This must be a result of the interpreter I was using, not intentional!

Reply:
hm, not the cause of mypy installation, I'll look into this now and see what went wrong.

Reply:
I have pushed another commit for you to review :)

Reply:
MIght be ye olde `--keep-outdated` flag to `pipenv install` that's missing, `pipenv` likes sneaky side effects :)

Reply:
yep, that did it. I think there is less diff without the markers. These 3 can be squashed. 3.7 is still failing regarding `typing-six` though, but I don't think it's because of this change.

Reply:
Nice, thanks for digging in!

I'm trying to verify the Travis CI jobs are actually running / passing, but I'm not seeing any integrated GitHub widgets/components like I was expecting. Are you able to see the test status / logs / coverage results / etc? 

I'm wondering if the CI integration needs to be updated after the [migration from travis-ci.org to travis-ci.com](https://blog.travis-ci.com/2021-05-07-orgshutdown)

Reply:
@bryant-finney No probs!

I have the jobs saved, but the last few runs I've just tested with tox - I'm out of credits for Travis - and the tests have been on my fork. Not sure how the plan looks for this repo in particular.

I'm considering migrating the travis yaml file over to GH actions to view builds on my fork.

If you would like the logs for what I did collect I can email them your way.

Reply:
Gotcha, yeah I was also considering forking to GitLab. I have a repo somewhere that uses `tox` with Docker executors for its CI jobs, but right now I'm not sure how to run tests in a Windows environment using GitLab. Does GitHub actions support windows runners?

Reply:
From what I know at this moment, GH actions should be able to do what Travis can do, I'm curious to look into it, but so far I've only used it for ubuntu builds. The good thing about actions is its integrated right into the repository, and widgets aren't a problem like you mentioned earlier.

Reply:
@jshwi 

Nice; I looked into it a little the other day. From what I understand, GitHub uses Azure-based infrastructure; so I would be very surprised if GitHub actions was unable to accommodate. 

While looking, I saw that GitLab recently started providing windows executors as well. Since I'm already comfortable with GitLab pipelines, I went ahead and forked this project over to [bryant.finney / pipenv-setup](https://gitlab.com/bryant.finney/pipenv-setup); that said, to me it seems better to use GitHub actions long-term.

So why don't we work on this in parallel? If CI/CD w/GitHub goes as quickly as one might hope, we can just drop the GitLab fork; otherwise, we can use the GitLab fork to run CI until we finish the GitHub actions integration. Does that sound reasonable to you?